source/leuven/wice_advanced_guide.rst

```diff
@@ -9,15 +9,68 @@ wICE advanced guide
 Compiling software
 ------------------
 
-The wICE compute nodes feature Intel processors nicknamed IceLake, which are
-the successors of the SkyLake and CascadeLake architectures which you can find
-on Genius. Although architectural differences are rather small, it is highly
-recommended to compile a version of your code specifically for wICE. A good approach
-to compile on any cluster (such as wICE) is to launch an interactive job (with the
-``srun`` command).
-
-Many dependencies you might need are centrally installed. The modules that are
-optimized for wICE are available when the appropriate
+Compared to the SkyLake and CascadeLake CPUs on Genius, the wICE nodes
+feature more recent CPU models such as Intel IceLake, Intel Sapphire Rapids
+and AMD Genoa. While architectural differences between SkyLake and CascadeLake
```

Comment:
I think this section and the next one can be expanded a bit. Of course it will be a balancing act to provide all useful information without overwhelming the users. Here are two points I would like to add:
* All these CPUs are based on the same instruction set architecture (ISA), but differences arise because different CPUs add additional instructions to the common set. Whether or not a code can benefit from the additional instructions offered by a CPU, is case-specific and cannot be discussed here in general. One very important remark is that the compiler needs to support CPU-specific instructions, which is why choosing a recent compiler on recent hardware is crucial. As an example, the Advanced Matrix Extensions introduced in Sapphire Rapids are only supported by GCC starting from version 11. When using an older compiler (for instance GCC 10.3.0 included in `foss-2021a`), the resulting binary might not be optimized for the Sapphire Rapids architecture.
* Next to differences in the ISA, also the organization of memory access has to be taken into account. The Intel IceLake nodes have 2 NUMA domains, while the Intel Sapphire Rapids and AMD Genoa nodes have 8 NUMA domains. This can be important to consider for instance when pinning MPI processes to cores, but again it is not possible to give general advice because this is case-specific.

These remarks should maybe not be included as-is, but if you can add some of it to this page it would be nice I think.

Reply:
thanks, I've given this a try in d454b75660fc68a5e9f1c36d9c0f17957fc4237d and be5e608b63f43a8a22c3cba6c488ca3cf8c24673